rllib/models/tf/tf_distributions.py

```diff
@@ -279,13 +277,259 @@ def kl(self, other: "Distribution", **kwargs) -> TensorType:
 
     @staticmethod
     @override(Distribution)
-    def required_model_output_shape(
-        space: gym.Space, model_config: ModelConfigDict
-    ) -> Tuple[int, ...]:
-        # TODO: This was copied from previous code. Is this correct? add unit test.
-        return (int(np.prod(space.shape, dtype=np.int32)),)
+    def required_input_dim(space: gym.Space, **kwargs) -> int:
+        return int(np.prod(space.shape, dtype=np.int32))
 
     @classmethod
     @override(Distribution)
     def from_logits(cls, logits: TensorType, **kwargs) -> "TfDeterministic":
         return TfDeterministic(loc=logits)
+
+
+@DeveloperAPI
+class TfMultiCategorical(Distribution):
+    """MultiCategorical distribution for MultiDiscrete action spaces."""
+
+    @override(Distribution)
+    def __init__(
+        self,
+        categoricals: List[TfCategorical],
+    ):
+        super().__init__()
+        self.cats = categoricals
+
+    @override(Distribution)
+    def sample(self) -> TensorType:
+        arr = [cat.sample() for cat in self.cats]
+        sample_ = tf.stack(arr, axis=1)
+        return sample_
+
+    @override(Distribution)
+    def rsample(self, sample_shape=()):
+        # TODO (Kourosh) Implement Categorical sampling using grad-passthrough trick.
```

Comment:
This TODO is a copy of the todo found above, in the categorical distribution.
I'm not adding the code here because we can't test it without `Categorical.rsample()`.